building/trusted_applications.rst

```diff
@@ -373,35 +373,48 @@ third step the binary and its signature are stitched together into the full TA.
 Offline Signing of TAs
 ======================
 
-The TA dev kit does sign an application as last step of the linking process. For
-example, the file ``ta/arch/arm/link.mk`` in the :ref:`optee_os` source tree
-contains the statement
+There are two types of TAs that can be signed offline. The in-tree TAs, which come with the OP-TEE
+OS (for example the ``pkcs11`` TA) and are generated during the compilation of the TA DEV KIT. The
+second type are any external TAs coming from the user. In both cases however, the signing process
+is the same.
+
+Offline signing is done with the following sequence of steps:
+
+0. (Preparation) Generate a 2048 bit RSA key for signing in a secure, offline
```

Comment:
about offline environment -- so this could even be online environment from code compilation point of view it is external.

Perhaps something like:
```
0. (Preparation) Generate RSA private key for signing in a secure environment and extract the public key. For example:
```